Marlin/src/HAL/HAL_STM32/HAL_timers_STM32.h

```diff
@@ -61,19 +61,7 @@
     #define TEMP_TIMER 2
   #endif
 
-#elif defined STM32F4xx
-
-  #define HAL_TIMER_RATE (F_CPU/2) // frequency of timer peripherals
-
-  #ifndef STEP_TIMER
-    #define STEP_TIMER 5
-  #endif
-
-  #ifndef TEMP_TIMER
-    #define TEMP_TIMER 7
-  #endif
-
-#elif defined STM32F7xx
+#elif defined(STM32F4xx) || defined(STM32F7xx) || defined(STM32H7xx)
 
   #define HAL_TIMER_RATE (F_CPU/2) // frequency of timer peripherals
```

Comment:
Note that depending on your timer choices., this may not be correct. I need to make a generic function which uses GetClockConfig to derive the prescalar, then calculates the correct work, but it's on the TODO list. If you already knew that, sorry.